### PR TITLE
implement photos

### LINKED ISSUE #21: photos - paths: figure out how to store photos and their paths. hash filenames? store in folders named afterusername? etc

### PR DESCRIPTION
**Pull request by travisCarson**
## Description

add ability to select a photo and upload 
add storage and lookup mechanisms
add photo rendering to rental items
## Related Issue

closes #21 
## How Has This Been Tested?

clicking through
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
## Checklist:

- [x](lol) My code follows the code style of this project. (lol)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,10 +7,13 @@
     "ejs": "^2.4.1",
     "express": "^4.10.1",
     "express-session": "^1.13.0",
+    "fs-extra": "^0.28.0",
```
Comment (willfulbard) [inline package.json:10]: Is fs-extra, multiparty, or ng-file-upload all used on the front end only, or are these backend as well?

File: client/app/services/services.js [hunk 2]
```diff
@@ -17,17 +17,54 @@ angular.module('e-Commer.services', [])
     }
   })
    
-  .factory('Item', function ($http) {
+  .factory('Item', function ($http, Upload) {
     return {
-      // add an item from a user
+
```
Comment (willfulbard) [inline client/app/services/services.js]: Is this zombie-code around for a reason?  I'd rather have it removed

File: server/controllers/itemController.js [hunk 3]
```diff
@@ -1,28 +1,65 @@
 // Sylvia
 var Items = require('../models/itemModel.js');
+var fs = require('fs-extra');
+var mkdirp = require('mkdirp');
```
Comment (willfulbard) [inline server/controllers/itemController.js:5]: I don't see mkdirp in the package.json, is this an included package or should it be added to the package.json file?

File: server/schemas.sql [hunk 4]
```diff
@@ -7,6 +7,7 @@ CREATE TABLE items (
   id int NOT NULL AUTO_INCREMENT,
   name varchar(50) NOT NULL,
   description varchar(200)  NOT NULL,
+  photo varchar(120) NOT NULL DEFAULT 'https://upload.wikimedia.org/wikipedia/commons/2/25/Icon-round-Question_mark.jpg',
   price int NOT NULL,
```
Comment (willfulbard) [inline server/schemas.sql:11]: Can we not hotlink this file?

File: package.json (hunk 1 above)
Comment (travisCarson) [inline package.json:10]: fs-extra, multiparty = backend.
ng-file-upload = frontend

File: client/app/services/services.js (hunk 2 above)
Comment (travisCarson) [inline client/app/services/services.js]: removed

File: server/controllers/itemController.js (hunk 3 above)
Comment (travisCarson) [inline server/controllers/itemController.js:5]: i've added it just in case

File: server/schemas.sql (hunk 4 above)
Comment (travisCarson) [inline server/schemas.sql:11]: done.

**Pull request merged**